### PR TITLE
Allow lens-5.3, QuickCheck-2.15; bump CI

### PR DESCRIPTION
**Pull request by andreasabel**
CI runs here: https://github.com/andreasabel/snap/pull/1

Published as revision 2: snap-1.1.3.3~2

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -92,38 +94,39 @@ jobs:
     strategy:
       matrix:
         include:
-          # GHC version must match https://www.stackage.org/nightly
+          # GHC version must match resolver
```
Comment (cydparser) [inline .github/workflows/ci.yml:97]: The comment is misleading. The nightly resolver is always used (see line 123).

**Review by cydparser — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (andreasabel) [reply .github/workflows/ci.yml:97]: Apologies, I overlooked line 123 `stack config set resolver nightly`.
Also, I do not know what it is doing.
The `stack.yaml` defines a very specific resolver for the project (now `nightly-2024-05-21`), so what effect does it have to globally set a resolver?
E.g. when I do this on my machine I get the response:
```
Selected resolver: nightly-2024-05-22.
~/.stack/global-project/stack.yaml has been updated.
```
This setting for the `global-project` should have no effect on the local project _that defines its own resolver_, does it?
So I guess the line 123 can be deleted.
But I am also happy to learn that I am wrong and what effect it actually has.

Comment (cydparser) [reply .github/workflows/ci.yml:97]: The specific resolver isn't actually used. It is only needed to get around the following error which occurs without a resolver:

```
$ stack config set resolver nightly

Error: [S-6602]
       Stack could not load and parse /home/cyd/snap/stack.yaml as a YAML configuraton file.

       While loading and parsing, Stack encountered the following error:

       Aeson exception:
       Error in $: failed to parse field "snapshot": keys ["snapshot","resolver"] not present
```

Line 123 updates the local resolver to the current nightly. The global behavior that you observed might be due to running the command without a `stack.yaml` in the current directory. I don't use stack---there might be a better way to say "always build with the latest nightly."

Here is the output from running the steps on a fresh checkout:

```
$ stack --version
Version 2.15.7, Git revision f590e0165b5ab92f5f7f87f8aa55852e1972ee25 x86_64 hpack-0.36.0

$ git clone --depth 1 git@github.com:snapframework/snap.git
$ cd snap
$ cat <<EOF > stack.yaml
packages:
  - '.'
resolver: nightly-2023-07-19
extra-deps:
  - 'clientsession-0.9.2.0'
  - 'snap-server-1.1.2.1'
EOF

$ stack config set system-ghc true --global
/home/cyd/.stack/config.yaml has been extended.

$ stack config set resolver nightly
Selected resolver: nightly-2024-05-23.
/home/cyd/snap/stack.yaml has been updated.

$ grep resolver stack.yaml
  resolver: nightly-2024-05-23
```

**Review by cydparser — CHANGES_REQUESTED** [review]
Revert changes to stack resolver.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (andreasabel) [reply .github/workflows/ci.yml:97]: Well, if you do not have a `stack.yaml` yet, then you might get the error you quote.  However, this workflow _first_ creates the `stack.yaml`.
As proven by this successful CI run, the line 123 is indeed superfluous: https://github.com/andreasabel/snap/actions/runs/9201233470/job/25309089198#step:4:7

**Comment by andreasabel** [comment]
> Revert changes to stack resolver.

See my comment.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (andreasabel) [reply .github/workflows/ci.yml:97]: I doubt this.  There was already a specific resolver written to the `stack.yaml` file.  I do not think that setting the resolver of the global-project will affect this local setting.
Do you have evidence (e.g. from prior CI runs) that actually the latest resolver was picked up always in presence of line 123?

Comment (andreasabel) [reply .github/workflows/ci.yml]: Apologies, you are absolutely right.
I did not expect `stack config set resolver` to act both on local and global configurations... (Weird UI design...)
Anyway, I restored the old logic.

But it seems that maybe a matrix does not make much sense, if the resolver is anyway set to latest nightly and so the GHC version is fixed.  (Maybe that also confused me.)  
Wouldn't it be clearer to remove the matrix in this job?

Comment (cydparser) [reply .github/workflows/ci.yml]: No worries. Removing the matrix makes sense.

**Pull request merged**